zerver/lib/presence.py

```diff
@@ -172,26 +182,47 @@ def get_presence_dict_by_realm(
             "user_profile_id",
             "user_profile__enable_offline_push_notifications",
             "user_profile__date_joined",
+            "last_update_id",
         )
     )
+    # Get max last_update_id from the list.
+    if presence_rows:
+        last_update_id_fetched: Optional[int] = max(row["last_update_id"] for row in presence_rows)
+    else:
+        # If there are no results, that means that are no new updates to presence
+        # since what the client has last seen. Therefore, returning the same
+        # last_update_id that the client provided is correct.
+        last_update_id_fetched = last_update_id
 
-    return get_presence_dicts_for_rows(presence_rows, slim_presence)
+    return get_presence_dicts_for_rows(presence_rows, slim_presence), last_update_id_fetched
 
 
 def get_presences_for_realm(
-    realm: Realm, slim_presence: bool, requesting_user_profile: UserProfile
-) -> Dict[str, Dict[str, Dict[str, Any]]]:
+    realm: Realm,
+    slim_presence: bool,
+    last_update_id: Optional[int],
+    requesting_user_profile: UserProfile,
+) -> Tuple[Dict[str, Dict[str, Dict[str, Any]]], Optional[int]]:
     if realm.presence_disabled:
         # Return an empty dict if presence is disabled in this realm
-        return defaultdict(dict)
+        return defaultdict(dict), None
 
-    return get_presence_dict_by_realm(realm, slim_presence, requesting_user_profile)
+    return get_presence_dict_by_realm(
+        realm, slim_presence, last_update_id, requesting_user_profile=requesting_user_profile
+    )
 
 
 def get_presence_response(
-    requesting_user_profile: UserProfile, slim_presence: bool
+    requesting_user_profile: UserProfile, slim_presence: bool, last_update_id: Optional[int] = None
```

Comment:
I feel like we probably want to call this `last_update_id` parameter something like `already_fetched_last_update_id` in the internal code.

Reply:
Did some renaming around to have these internal code variables named `last_update_id_fetched_by_client` and `last_update_id_fetched_by_server`